solutions/silver/cf-803D.mdx

```diff
@@ -0,0 +1,109 @@
+---
+id: cf-803D
+source: CF
+title: Magazine Ad
+author: Alex Du
+---
+
+<Spoiler title="Hint 1">
+
+Is there a way to format an ad of a given width using less than or equal to the maximum number of lines?
+
+</Spoiler>
+
+
+<Spoiler title="Hint 2">
+
+If you know that some width is/isn't possible to format an ad with, what can you deduce from this?
+
+</Spoiler>
+
+<Spoiler title="Solution">
+
+## Explanation
+
+[Official Analysis](https://codeforces.com/blog/entry/51768)
+
+Is there a way to format an ad of a given width using less than or equal to the maximum number of lines?
+
+If we can format an ad using less than or equal to the maximum number of lines, we can binary search for the minimum width that works.
```

Comment:
would be awesome if u could say something like

note that if $x$ width works, then all $i < x$ also works, **so** we can binary search